### PR TITLE
Fix: LIVE-13547 Replace expo-random lib with expo-crypto

### PR DESCRIPTION
**Pull request by hzheng-ledger**
### ✅ Checklist

- [X] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [X] **Impact of the changes:** No impact
  - ...

### 📝 Description

Replace the expo-random lib with expo-crypto
Same PR has been created by Zondax : https://github.com/LedgerHQ/ledger-live/pull/7459

### ❓ Context

- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/LIVE-13547

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: libs/ledger-live-common/package.json [hunk 1]
```diff
@@ -193,8 +193,8 @@
     "@stricahq/typhonjs": "^2.0.0",
     "@taquito/ledger-signer": "^20.0.0",
     "@ton-community/ton-ledger": "^7.0.1",
-    "@ton/core": "^0.56.1",
-    "@ton/crypto": "3.2.0",
+    "@ton/core": "^0.56.3",
+    "@ton/crypto": "^3.3.0",
```
Comment (Justkant) [inline libs/ledger-live-common/package.json:197]:
```suggestion
    "@ton/crypto": "^3.3.0",
```

File: apps/ledger-live-mobile/package.json [hunk 2]
```diff
@@ -146,7 +146,7 @@
     "expo-keep-awake": "~12.8.2",
     "expo-modules-autolinking": "^1.10.2",
     "expo-modules-core": "^1.11.8",
-    "expo-random": "^13.6.0",
+    "expo-crypto": "^12.8.1",
```
Comment (Justkant) [inline apps/ledger-live-mobile/package.json:149]: Why not use the latest version (`13.0.2`) of `expo-crypto` ?

**Review by Justkant — COMMENTED** [review]
Small details

File: libs/ledger-live-common/package.json (hunk 1 above)
Comment (hzheng-ledger) [reply libs/ledger-live-common/package.json:197]: I'll fix it. Thanks.

File: apps/ledger-live-mobile/package.json (hunk 2 above)
Comment (hzheng-ledger) [reply apps/ledger-live-mobile/package.json:149]: good point. 
In fact, I just take the same PR as zondax did. https://github.com/LedgerHQ/ledger-live/pull/7459/files
I will try with your suggestion, thanks

File: libs/ledger-live-common/package.json (hunk 1 above)
Comment (hzheng-ledger) [reply libs/ledger-live-common/package.json]: fixed

File: apps/ledger-live-mobile/package.json (hunk 2 above)
Comment (hzheng-ledger) [reply apps/ledger-live-mobile/package.json:149]: In fact, 
in the new version of expo-crypto
the https://github.com/expo/expo/blob/main/packages/expo-crypto/android/build.gradle
is not compatible with our current android SDK version and it causes an error in CI.

it seems that the "useDefaultAndroidSdkVersions" needs a higher version of android SDK version.

Comment (Justkant) [reply apps/ledger-live-mobile/package.json:149]: Okay too bad maybe we'll need to update that at some point with the next RN update

**Review by Justkant — APPROVED** [review]

**Pull request merged**